src/Timers.cpp

```diff
@@ -410,8 +859,134 @@ PVR_ERROR Timers::UpdateTimer(const PVR_TIMER &timer)
   return PVR_ERROR_NO_ERROR; 
 }
 
+PVR_ERROR Timers::UpdateAutoTimer(const PVR_TIMER &timer)
+{
+  unsigned int i=0;
+  while (i<m_autotimers.size())
+  {
+    if (m_autotimers.at(i).iClientIndex == timer.iClientIndex)
+      break;
+    else
+      i++;
+  }
+
+  AutoTimer timerToUpdate = m_autotimers.at(i);
```

Comment:
What happens if you don't find the timer? You will return the last elem in your timers vector. Is this intened?

I would rewrite this like:

```C++
const auto it = std::find_if(
      m_autotimers.cbegin(),
      m_autotimers.cend(),
      [timer](const AutoTimer& autoTimer)
    {
      return autoTimer.iClientIndex  == timer.iClientIndex;
    });

    if (it != m_autotimers.cend())
    {
```